apps/reference/docs/guides/auth/server-side-rendering.mdx

```diff
@@ -0,0 +1,190 @@
+
+As seen from the authentication flow, the first request made by the browser to
+your app's server after user login **does not contain any information about the
+user**.
+
+You need to make sure that the redirect route works without any server-side
+rendering. Other routes requiring authorization do not have the same
+limitation, provided you give your server the access and refresh tokens.
+
+This is traditionally done by setting cookies. Here's an example you
+can add to the root of your application:
+
+```typescript
+supabase.auth.onAuthStateChange((event, session) => {
+  if (event === "SIGNED_OUT" || event === "USER_DELETED") {
+    // delete cookies on sign out
+    const expires = new Date(0).toUTCString();
+    document.cookie = `my-access-token=; path=/; expires=${expires}; SameSite=Lax; secure`;
+    document.cookie = `my-refresh-token=; path=/; expires=${expires}; SameSite=Lax; secure`;
+  } else if (event === "SIGNED_IN" || event === "TOKEN_REFRESHED") {
+    const maxAge = 100 * 365 * 24 * 60 * 60; // 100 years, never expires
+    document.cookie = `my-access-token=${session.access_token}; path=/; max-age=${maxAge}; SameSite=Lax; secure`;
+    document.cookie = `my-refresh-token=${session.refresh_token}; path=/; max-age=${maxAge}; SameSite=Lax; secure`;
+  }
+})
+```
+
+This uses the standard
+[`document.cookie` API](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie)
+to set cookies on all paths of your app's domain. All subsequent requests
+made by the browser to your app's server include the `my-access-token` and
+`my-refresh-token` cookies (the names of the cookies and additional
+parameters can be changed).
+
+In your server-side rendering code you can now access user and session
+information:
+
+
+```typescript
+const refreshToken = req.cookies['my-refresh-token'];
+
+if (refreshToken) {
+  await supabase.auth.setSession(refreshToken);
+} else {
+  // make sure you handle this case!
+  throw new Error("User is not authenticated.")
+}
+
+// returns user information
+await supabase.auth.getUser();
+```
+
+Use `setSession(refreshToken)` instead of `getUser(accessToken)` as refresh
+tokens are long-lived credentials that provide user information even if the user
+has not used your app in a long time.
+
+Even though refresh tokens are long-lived, they is no guarantee that a user
+has an active session. They may have logged out and your application failed to
+remove the `my-refresh-token` cookie, or some other failure occurred that left
+a stale refresh token in the browser.
+
+A good practice is to handle unauthorized errors by deferring to render the page
+in the browser instead of in the server. Some user information is contained in the
+access token though, so in certain cases, you may be able to use this potentially
+stale information to render a page.
+
+## Frequently Asked Questions
+
+### How do I make the cookies `HttpOnly`?
+
+This is not necessary. Both the access token and refresh token are designed to
```

Comment:
Maybe instead of saying it's not necessary, we should phrase it as it's not possible for now because in order for the client library to work, it needs to be able to access the access token and refresh token. Eventually, we would default to using strict http-only cookies instead of local storage.

Reply:
Ah actually we won't be able to use http-only cookies in a general sense. The access and refresh token have to be exchanged with the JavaScript / Kotlin / Swift / Dart (Flutter) as not all apps are browser apps (even though they use web technologies). Thus the sign-in flow has to somehow give access to the client code to the refresh token. Placing it as http-only no longer makes sense in such a situation.

Reply:
I think in the long-run, we can actually move the refresh token to use http-only cookies. If it's used in a non-browser context, then they would be able to read the cookie like a normal header right?